core/server/proxy/src/main/java/alluxio/proxy/s3/S3RangeSpec.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.proxy.s3;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This class is Range Spec for Amazon S3 API.
+ */
+public class S3RangeSpec {
+  private static final Pattern RANGE_PATTERN = Pattern.compile("^bytes=(\\d*)-(\\d*)$");
+
+  public static final S3RangeSpec INVALID_S3_RANGE_SPEC = new S3RangeSpec(false);
+
+  /* Does the range spec is valid */
+  private boolean mIsValid;
+  /* Does the range spec refers to a suffix of object */
+  private boolean mIsSuffixLength;
+  /* Start offset specified in range spec */
+  private long mStart;
+  /* End offset specified in range spec */
+  private long mEnd;
+
+  /**
+   * @param isValid the spec is valid or not
+   */
+  public S3RangeSpec(boolean isValid) {
+    mIsValid = isValid;
```

Comment:
I would make this constructor protected to prevent misuse, but still have available in tests.